pointmatcher_ros/src/StampedPointCloud.cpp

```diff
@@ -171,16 +171,15 @@ bool StampedPointCloud::filterByDistance(const float distanceThreshold, const bo
 
 bool StampedPointCloud::filterByBoundingBox(const float xMin, const float xMax, const float yMin, const float yMax, const float zMin,
                                             const float zMax, const bool keepInside) {
-  PmPointCloudFilter* boundingBoxFilter = Pm::get().DataPointsFilterRegistrar.create(
-      "BoundingBoxDataPointsFilter", {{"xMin", PointMatcherSupport::toParam(xMin)},
-                                      {"xMax", PointMatcherSupport::toParam(xMax)},
-                                      {"yMin", PointMatcherSupport::toParam(yMin)},
-                                      {"yMax", PointMatcherSupport::toParam(yMax)},
-                                      {"zMin", PointMatcherSupport::toParam(zMin)},
-                                      {"zMax", PointMatcherSupport::toParam(zMax)},
-                                      {"removeInside", PointMatcherSupport::toParam(!keepInside)}});
+  auto boundingBoxFilter = Pm::get().DataPointsFilterRegistrar.create("BoundingBoxDataPointsFilter",
```

Comment:
Is it wise to add auto when it is not clear what kind of type is being returned?

Reply:
Old libpointmatcher used to return a raw pointer, and after syncing upstream it returns a shared pointer. This is here just for compatibility.